sympy/categories/diagram_drawing.py

```diff
@@ -670,7 +670,7 @@ def _weld_triangle(tri, welding_edge, fringe, grid, skeleton):
         a = (a[0] + offset[0], a[1] + offset[1])
         b = (b[0] + offset[0], b[1] + offset[1])
 
-        fringe.extend([(a, target_cell), (b, target_cell)])
+        fringe.extend(((a, target_cell), (b, target_cell)))
```

Comment:
There are still quite a few changes like this. I think this would be better left unchanged or should be changed to
```python
fringe += [(a, target_cell), (b, target_cell)]
```
since that's the fastest version.

Reply:
Now, there are a few aspects here, just to be clear:
* If the content is compile-time constant?
* More than two terms?
* Some compile-time constants?

Reply:
If the tuple is entirely a compile time constant then a tuple expression like `(1, 2)` is efficient because it is *fully* precomputed. Otherwise I don't think tuples have any advantage in this situation. Creating a small tuple from a literal tuple expression should be just as fast as creating a small list and both should be just as fast when iterating.

Using `+=` will usually be faster than `extend` because operators are faster than methods in CPython. This will only be significant when extending by a small list.

Using a generator with `extend` is slower on a per-item basis because both `list.extend` and `list.__iadd__` have fast paths for concrete containers like list and tuple that can not be used with a generator.

Reply:
One could argue that creation of tuples is slightly faster, at least in terms of "python operations", and I've seen minor benefits even then. I guess there is a reason why the points are tuples and not lists. (Even though one can also argue that they should be tuples as they do not need to be manipulated.)

I'll get rid of the generator extends I've added, but here I do actually only see benefits (unless we should change all `extend` to `+=` which I will not do currently).